projects/step-frontend/src/lib/modules/timeseries/execution-page/execution-page.component.ts

```diff
@@ -120,41 +174,56 @@ export class ExecutionPageComponent implements OnInit, OnDestroy {
     if (oldInterval.value === newInterval.value) {
       return;
     }
+    // this.updatingSubscription.unsubscribe();
+    this.terminator$.next();
     if (newInterval.value) {
-      this.updatingSubscription.unsubscribe();
-      this.triggerNextUpdate(newInterval.value, of(undefined));
+      let delay = oldInterval.value === 0 ? 0 : newInterval.value;
```

Comment:
Please use brackets so it is immediately obvious in which order there operations are done (also most compilers don't guarantee a fixed order)